File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.swift
import Foundation


func solution(_ nums:[Int]) -> Int {
    var sum = 0
    var numPrimeNumber = 0
    
    for i in 0..<nums.count-2{
        for j in i+1..<nums.count-1{
            for k in j+1..<nums.count{
                sum = nums[i]+nums[j]+nums[k]
                if (isPrime(sum)){
                    numPrimeNumber += 1
                }
            }
        }
    }
    
    return numPrimeNumber
}

func isPrime(_ num: Int)->Bool{
    var sqrtNum = Int(sqrt(Double(num)))
    for i in 2...sqrtNum{
      
        if(num%i == 0){
            return false
        }
    }
    return true
}

File: 프로그래머스/Lv.1/12977. 소수 만들기/소수 만들기.py
from solution import *

def solution(nums):
    answer = 0
    n = len(nums)
    for i in range(0,n-2):
        for j in range(i+1,n-1):
            for k in range(j+1,n):
                sum = nums[i] + nums[j] + nums[k]
                print("합",sum)
                if(isSosu(sum)):
                    answer += 1
    
    

    return answer

def isSosu(num):
    if num == 1 | num == 2:
        return True
    
    for i in range(2,num-1):
        if num%i == 0:
            return False
    
    return True
        